File: gui/Text.py
class Text:

    def __init__(self,canvas,x,y,font,size,weight,color,prefix,suffix,text):
        self.canvas=canvas
        self.font=font
        self.size=size
        self.weight=weight
        self.color=color
        self.prefix=str(prefix)
        self.suffix=str(suffix)
        
        self.id=canvas.create_text(x,y,text=prefix+text+suffix,font=self.font+" "+str(self.size)+" "+self.weight,fill=self.color)

    def setText(self,text):
        self.canvas.itemconfig(self.id,text=self.prefix+str(text)+self.suffix)

    def setColor(self,color):
	self.canvas.itemconfig(self.id,fill=color)

File: src/frontend/components/text.js
class Text {
  constructor(args) {
    this.element = document.getElementById(args.id);
    this.prefix = args.prefix || "";
    this.suffix = args.suffix || "";

    this.element.innerHTML = this.prefix + args.value + this.suffix;
    this.element.style.fontSize = args.size + "vw";
    this.element.style.fontFamily = args.font || "arial";
    this.element.style.fontWeight = args.weight || "bold";
    this.element.style.fontStyle = args.style || "";
    this.element.style.textAlign = args.textAlign || "";
  }

  refresh(value) {
    this.element.innerHTML = this.prefix + value + this.suffix;
  }

  setSuffix(suffix) {
    if (suffix != null) {
      this.suffix = suffix;
    }
  }

  setColor(color) {
    this.element.style.color = color;
  }
}
